packages/shared/lib/core/layer-2/actions/fetchSelectedAccountLayer2Balance.ts

```diff
@@ -72,3 +86,28 @@ async function getSelectedAccountLayer2BalanceForAddress(
         console.error(error)
     }
 }
+
+async function getSelectedAccountLayer2Erc20BalancesForAddress(
+    selectedAccountEvmAddress: string,
+    chain: IChain
+): Promise<{ balance: number; tokenId: string }[] | undefined> {
+    try {
+        const chainId = chain.getConfiguration().chainId
+        const trackedTokens = get(selectedAccount)?.trackedTokens?.[chainId] ?? []
+        const erc20TokenBalances = []
+        for (const erc20Address of trackedTokens) {
+            const contract = chain?.getContract(ContractType.Erc20, erc20Address)
+            if (contract) {
+                const coinType = chain?.getConfiguration().coinType
+                if (coinType) {
+                    const rawBalance = await contract.methods.balanceOf(selectedAccountEvmAddress).call()
+                    erc20TokenBalances.push({ balance: rawBalance, tokenId: erc20Address })
+                }
+            }
```

Comment:
im not a big fan of this nested `if`s. i suggest:

```suggestion
            const contract = chain?.getContract(ContractType.Erc20, erc20Address)
            const coinType = chain?.getConfiguration().coinType
            if (!contract || !coinType) {
                continue
            }
            const rawBalance = await contract.methods.balanceOf(selectedAccountEvmAddress).call()
            erc20TokenBalances.push({ balance: rawBalance, tokenId: erc20Address })
```